static/app/components/events/interfaces/spans/spanTreeModel.tsx

```diff
@@ -301,57 +314,219 @@ class SpanTreeModel {
       }
     }
 
-    const {descendants} = (hideSpanTree ? [] : descendantsSource).reduce(
-      (
-        acc: {
-          descendants: EnhancedProcessedSpanType[];
-          previousSiblingEndTimestamp: number | undefined;
-        },
-        span,
-        index
-      ) => {
-        acc.descendants.push(
-          ...span.getSpansList({
-            operationNameFilters,
-            generateBounds,
-            treeDepth: shouldHideSpanOfGroup ? treeDepth : treeDepth + 1,
-            isLastSibling: index === lastIndex,
-            continuingTreeDepths: descendantContinuingTreeDepths,
-            hiddenSpanGroups,
-            spanGroups: new Set(childSpanGroup),
-            filterSpans,
-            previousSiblingEndTimestamp: acc.previousSiblingEndTimestamp,
-            event,
-            isOnlySibling: descendantsSource.length === 1,
-            spanGrouping: shouldGroup
-              ? [...(spanGrouping ?? []), wrappedSpan]
-              : undefined,
-            toggleSpanGroup: isNotLastSpanOfGroup
-              ? toggleSpanGroup === undefined
-                ? this.toggleSpanGroup
-                : toggleSpanGroup
-              : undefined,
-            showSpanGroup: isNotLastSpanOfGroup
-              ? toggleSpanGroup === undefined
-                ? this.showSpanGroup
-                : showSpanGroup
-              : false,
-            addTraceBounds,
-            removeTraceBounds,
-          })
-        );
-
-        acc.previousSiblingEndTimestamp = span.span.timestamp;
-
-        return acc;
-      },
-      {
-        descendants: [],
-        previousSiblingEndTimestamp: undefined,
+    let descendants: EnhancedProcessedSpanType[];
+
+    if (isAutogroupSiblingFeatureEnabled) {
+      // Check if the descendants in this span have consecutive similar siblings, and group them
+      const groupedDescendants: SpanTreeModel[][] = [];
+      if (descendantsSource?.length >= MIN_SIBLING_GROUP_SIZE) {
+        let prevSpanModel = descendantsSource[0];
+        let currentGroup = [prevSpanModel];
+
+        for (let i = 1; i < descendantsSource.length; i++) {
+          const currSpanModel = descendantsSource[i];
```

Comment:
@0Calories This could be simplified with `for ... of` loop:

```suggestion
        for (const currSpanModel of descendantsSource) {
```

Reference: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/for...of

Reply:
@dashed I opted to use a for loop here since I had to start from index 1. I could switch to a `for ... of` loop but then I'd need an additional if statement within it to skip the first element.